applications/vanilla/controllers/class.discussioncontroller.php

```diff
@@ -82,7 +82,9 @@ public function index($DiscussionID = '', $DiscussionStub = '', $Page = '') {
         // Check permissions.
         $Category = CategoryModel::categories($this->Discussion->CategoryID);
         $this->permission('Vanilla.Discussions.View', true, 'Category', val('PermissionCategoryID', $Category, -1));
-        $this->setData('CategoryID', $this->CategoryID = $this->Discussion->CategoryID, true);
+        if (c('Vanilla.Categories.Use', true)) {
+            $this->setData('CategoryID', $this->CategoryID = $this->Discussion->CategoryID, true);
```

Comment:
I'm not sure about the `$this->CategoryID = null;`. It is quite verbose since it is defined in as a class var and never set, so it is already null. Changing this:

~~~
    /** @var array Unique identifier. */
    public $CategoryID;
~~~

to that:

~~~
    /** @var array Unique identifier. */
    public $CategoryID = null;
~~~

and leaving out the "else..." part I've added would be something I prefer. It is not as self explaining as the code above, but setting a variable to a "value" (null) it already has does feel awkward to me, too.